src/main/java/bio/terra/service/filedata/google/firestore/FireStoreBatchQueryIterator.java

```diff
@@ -59,26 +63,20 @@ public FireStoreBatchQueryIterator(Query baseQuery, int batchSize) {
     logger.info("Retrieving batch {} with batch size of {}", count, batchSize);
     count++;
 
-    for (int i = 0; i < RETRIES; i++) {
-      ApiFuture<QuerySnapshot> querySnapshot = query.get();
+    currentList =
+        fireStoreUtils.runTransactionWithRetry(
+            firestore,
+            xn -> {
+              ApiFuture<QuerySnapshot> querySnapshot = xn.get(query);
+              return querySnapshot.get().getDocuments();
+            },
+            "getBatch",
+            " Retrieving batch " + count + " with batch size of " + batchSize);
```

Comment:
I didn't notice this before, but we shouldn't rely on the caller to pad an argument with a leading space. Looking at how this message is used in `runTransactionWithRetry()` I don't think the space is needed at all.